File: code/backend/src/main/java/com/mysterytrip/controller/PlaceController.java
package com.mysterytrip.controller;

import com.mysterytrip.entity.Place;
import com.mysterytrip.repository.PlaceRepository;
import org.springframework.web.bind.annotation.*;

import java.util.List;

@RestController
@RequestMapping("/api/places")
public class PlaceController {

    private final PlaceRepository repository;

    public PlaceController(PlaceRepository repository) {
        this.repository = repository;
    }

    /** GET /api/places?type=LODGING|RESTAURANT|ATTRACTION (type 없으면 전체) */
    @GetMapping
    public List<Place> getPlaces(@RequestParam(required = false) Place.PlaceType type) {
        return type == null ? repository.findAll() : repository.findByType(type);
    }
}

File: code/backend/src/main/kotlin/com/trevit/controller/PlaceController.kt
package com.trevit.controller

import com.trevit.entity.Place
import com.trevit.repository.PlaceRepository
import org.springframework.web.bind.annotation.GetMapping
import org.springframework.web.bind.annotation.RequestMapping
import org.springframework.web.bind.annotation.RequestParam
import org.springframework.web.bind.annotation.RestController

@RestController
@RequestMapping("/api/places")
class PlaceController(private val repository: PlaceRepository) {

    /** GET /api/places?type=LODGING|RESTAURANT|ATTRACTION (type 없으면 전체) */
    @GetMapping
    fun getPlaces(@RequestParam(required = false) type: Place.PlaceType?): List<Place> =
        if (type == null) repository.findAll() else repository.findByType(type)
}
